container-engine-lib/lib/backend_impls/kubernetes/kubernetes_kurtosis_backend/engine_functions/create_engine.go

```diff
@@ -165,13 +179,34 @@ func CreateEngine(
 		}
 	}()
 
+	engineIngress, err := createEngineIngress(
+		ctx,
+		namespaceName,
+		engineAttributesProvider,
+		privateRESTAPIPortSpec,
+		kubernetesManager,
+	)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred creating the engine ingress")
+	}
+	var shouldRemoveIngress = true
+	defer func() {
```

Comment:
My reading on why we have this async (goroutine) part is this is a non-critical code path that if fails there is no big deal. So we can spawn off and try it.  But the `ERROR` log and `ACTION REQUIRED` in case of failure gives a conflicting reading.

Reply:
Those are not asyncs but defers: cleanup in case of an early exit due to an error.